lib/checkclass.cpp

```diff
@@ -659,12 +659,11 @@ std::vector<CheckClass::Usage> CheckClass::createUsageList(const Scope *scope)
 
 void CheckClass::assignVar(std::vector<Usage> &usageList, nonneg int varid)
 {
-    for (Usage& usage: usageList) {
-        if (usage.var->declarationId() == varid) {
-            usage.assign = true;
-            return;
-        }
-    }
+    auto it = std::find_if(usageList.begin(), usageList.end(), [varid](const Usage& usage) {
```

Comment:
I would think that cbegin/cend is preferable.

Reply:
When called with const iterators, the returned iterator will also be const: https://godbolt.org/z/YTdjq9eE1